### PR TITLE
Feature delete session

### PR DESCRIPTION
**Pull request by webatspeed**
COBH-4018

### CODE REVIEW

File: api/userservice.yaml [hunk 1]
```diff
@@ -1173,6 +1173,37 @@ paths:
...
+          description: Session id
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        200:
+          description: OK - successful operation
+        400:
+          description: BAD REQUEST - invalid/incomplete request resp. provided password is wrong
```
Comment (maeventer) [inline api/userservice.yaml]: There is no password provided in the enpoint definition. Is the password parameter missing or should the description be adapted?

File: src/main/java/de/caritas/cob/userservice/api/deleteworkflow/service/DeleteUsersRegisteredOnlyService.java [hunk 2]
```diff
@@ -39,26 +40,28 @@
...
-    }
-
+  public void deleteUserAccountsTimeSensitive() {
+    var dateTimeToCheck = DateCalculator
+        .calculateDateInThePastAtMidnight(userRegisteredOnlyDeleteWorkflowCheckDays);
+    deleteUserAccountsBefore(dateTimeToCheck);
   }
 
-  private List<DeletionWorkflowError> deleteAskersAndCollectPossibleErrors() {
+  public void deleteUserAccountsTimeInsensitive() {
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/userservice/api/deleteworkflow/service/DeleteUsersRegisteredOnlyService.java:53]: Please add a javadoc comment for the public method. We are always adding javadocs on only public methods ;-)

**Review by maeventer — CHANGES_REQUESTED** [review]

File: src/main/java/de/caritas/cob/userservice/api/deleteworkflow/service/DeleteUsersRegisteredOnlyService.java [hunk 3]
```diff
@@ -39,26 +40,28 @@
...
+    var startOfTomorrow = LocalDateTime.now().with(LocalTime.MIDNIGHT).plusDays(1);
```
Comment (maeventer) [inline src/main/java/de/caritas/cob/userservice/api/deleteworkflow/service/DeleteUsersRegisteredOnlyService.java]: For the current time we use always the ZoneOffset UTC. Therefore there is a static method "nowInUtc()"

File: api/userservice.yaml (hunk 1 above)
Comment (webatspeed) [reply api/userservice.yaml]: Apparently, the description is wrong unless there is internal auth somehow in addition to the Keycloak token. Is there? Changed the description.

File: src/main/java/de/caritas/cob/userservice/api/deleteworkflow/service/DeleteUsersRegisteredOnlyService.java (hunk 3 above)
Comment (webatspeed) [reply src/main/java/de/caritas/cob/userservice/api/deleteworkflow/service/DeleteUsersRegisteredOnlyService.java]: Changed. But in the existing implementation inside DateCalculator.calculateDateInThePastAtMidnight it is also not used. Should that be changed?

Comment (maeventer) [reply src/main/java/de/caritas/cob/userservice/api/deleteworkflow/service/DeleteUsersRegisteredOnlyService.java]: Oh yes please change it there too. You have eagle eyes :-)

**Review by maeventer — APPROVED** [review]

**Pull request merged**